unittests/internaltests.py

```diff
@@ -674,6 +674,36 @@ def _call_pkgbin(self, args, env=None):
                         for lib in ('pthread', 'm', 'c', 'dl', 'rt'):
                             self.assertNotIn(f'lib{lib}.a', link_arg, msg=link_args)
 
+    def test_program_version(self):
+        if is_windows():
+            return
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            script_path = Path(tmpdir) / 'script'
+            script_path.write_text('#!/bin/sh\necho "$1"\n', encoding='utf-8')
+            script_path.chmod(0o755)
+
+            for output, expected in {
```

Comment:
Wonder if any of the other examples from https://wiki.gentoo.org/wiki/Project:Perl/Version-Scheme break.

Reply:
Not our problem, perl software with bin entrypoints is responsible for setting a sensible version number that they advertise for other projects to check for and rely on.

Reply:
I also don't think they do break -- all of the examples contain dots, so we'd continue to parse them correctly.